### PR TITLE
Adding terra-app-delegate, terra-modal-manager, terra-slide-group

### PR DESCRIPTION
**Pull request by tbiethman**
### Summary
Adding the following packages to facilitate future application development:
* terra-app-delegate
* terra-modal-manager
* terra-slide-group

These are transplants from terra-clinical. The terra-clinical packages will be removed upon release of these new packages.

### CODE REVIEW

**Comment by bjankord** [comment]
Hey @tbiethman there was a issue with the heroku deployment and it did not build out successfully for this PR. Now Heroku is saying "Name is already taken" for this PR's deployment. Would you mind branching off of this branch and then submitting that as a new PR. That should get around the heroku deployment issue.

File: packages/terra-app-delegate/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Terra App Delegate
+
+
+[![NPM version](http://img.shields.io/npm/v/terra-app-delegate.svg)](https://www.npmjs.org/package/terra-app-delegate)
+[![Build Status](https://travis-ci.org/cerner/terra.svg?branch=master)](https://travis-ci.org/cerner/terra)
```
Comment (Matt-Butler) [inline packages/terra-app-delegate/README.md]: [Build Status](https://travis-ci.org/cerner/terra-core)

File: packages/terra-app-delegate/package.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "terra-app-delegate",
+  "main": "lib/app-delegate/AppDelegate.js",
+  "version": "0.1.1",
```
Comment (Matt-Butler) [inline packages/terra-app-delegate/package.json]: Should we start this at 0.1?

File: packages/terra-app-delegate/package.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  "description": "An object that provides a set of functions used to facilitate communcation between Containers and their consuming Containers/Applications",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/cerner/terra.git"
```
Comment (Matt-Butler) [inline packages/terra-app-delegate/package.json]: "git+https://github.com/cerner/terra-core.git"

File: packages/terra-app-delegate/README.md (hunk 1 above)
Comment (tbiethman) [inline packages/terra-app-delegate/README.md]: Doh

File: packages/terra-app-delegate/package.json (hunk 3 above)
Comment (tbiethman) [inline packages/terra-app-delegate/package.json]: Doh Doh

**Comment by tbiethman** [comment]
New branch was created, I'll be updating things there #520

File: packages/terra-app-delegate/package.json (hunk 3 above)
Comment (Matt-Butler) [inline packages/terra-app-delegate/package.json]: There are a couple other paths below that just need core added to them

File: packages/terra-modal-manager/package.json [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    "terra-app-delegate": "^0.x",
+    "terra-mixins": "^1.0.0"
+  },
+  "dependencies": {
+    "classnames": "^2.2.5",
+    "prop-types": "^15.5.8",
+    "terra-base": "^0.x",
+    "terra-slide-group": "^0.x",
+    "terra-mixins": "^1.0.0",
+    "terra-modal": "0.3.0",
```
Comment (Matt-Butler) [inline packages/terra-modal-manager/package.json]: unconstrain?

Comment (tbiethman) [inline packages/terra-modal-manager/package.json]: Sounds like a plan

**Pull request closed without merging**